### ISSUE TITLE
NPM support

### ISSUE THREAD

**Issue by WDShannon** [iamisti/mdDataTable#95]
Could you please add this to the NPM repository so we can install it from there?  We do not use bower and would love for this to fit into our development eco system.

**Comment by iamisti** [comment]
The support for npm is there. The package name is `md-data-table` (due the naming convention restrictions on npm)
https://github.com/iamisti/mdDataTable/blob/master/package.json

**Comment by WDShannon** [comment]
Hey thanks for the reply.  I do see it in NPM now; I just assumed since only bower was listed in the documentation that its all that was offered.

**Comment by iamisti** [comment]
You right, I leave this ticket open to remind myself to extend the documentation.

**Comment by jvieyra92** [comment]
This is perfect! I'm new to npm, so I'm not sure how to install mdDataTable through the terminal. Can you update the readme to say how to install through npm? Thanks in advance.

**Comment by surfjedi** [comment]
+1 I could use this too - i think it will also need an index.js file which requires all the other files

**Comment by surfjedi** [comment]
@WDShannon Just actually updated some deps and the reademe to install with npm and use with webpack for example.  this the PR https://github.com/iamisti/mdDataTable/pull/121

**Comment by isarvalley** [comment]
I'm very new to AngularJS and all the related tools, but I like to use your promising mdDataTable. At the moment I am stuck with installing this module via jspm and loading all required modules.

Maybe you could merge surfjedi's pull request and update the npm reference? At https://www.npmjs.com/package/md-data-table is current version still 1.4.2.

**Comment by surfjedi** [comment]
@isarvalley in the mean time you can always install from my repo with npm till this is merged and up on npm

**Comment by isarvalley** [comment]
@surfjedi thank you for your reply. I tried to use your repo, but as I am using jspm as packet manager I got an error, that the dependencies have to be of form owner/repo. 

So I made my own fork and changed the dependencies. After that it seems to install the package correctly. But when I try to import the package with ES6 in my app.js with `import  mdDataTable from "md-data-table";` it tries to load `GET http://localhost:8080/md-data-table.js 404 (Not Found)` with no success.  Do you have any clue for me?

**Comment by surfjedi** [comment]
Ahh.. i think I know the problem - "import 'angular-material-data-table';" not "md-data-table"...
Its because you need to import by the package name (angular-material-data-table). You want to import the whole package - as opposed to just one file.  And you'll need the CSS too - if it s with webpack -
require("style!css!../../../node_modules/angular-material-data-table/dist/md-data-table.min.css");
but you may change how you do that, I'm not familiar with jspm set ups yet.

Try that and let me know

**Comment by isarvalley** [comment]
Sorry, doesn't work either. Can you tell me where the package Name ist defined? I can't find any occurence of angular-material-data-table. Maybe you mixed something up with the implementation of Daniel Nagy? I saw you forked his Repo, too. And there I could find this String in the package.json.

**Comment by surfjedi** [comment]
Ooopss.. you're right that was my bad - I'll have to take a deeper look.

**Comment by isarvalley** [comment]
Ok, I sorted it out. It was my fault. I forked your branch into my git repo. And after installing it via jspm the dependency in package.json got renamed to `isarvalley/mdDataTable`. So I change the import to `import "isarvalley/mdDataTable";`After that the script loads but throws this error: `Controller 'mdtTable', required by directive 'mdtHeaderRow', can't be found!` I know there is already an issue addressing this problem, but none of the solutions described there works for me. Bad luck.
